### PR TITLE
Update base path

### PR DESCRIPTION
**Pull request by hummusonrails**
Refactor base path in gem to check if its within Rails and if the Rails configuration is defined. If yes, use that. If not, default to `.`.

### CODE REVIEW

File: lib/nexmo_markdown_renderer.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # frozen_string_literal: true
 GEM_ROOT = File.expand_path("../..", __FILE__)
+DOCS_BASE_PATH = (defined?(Rails) && Rails.configuration.docs_base_path) ? Rails.configuration.docs_base_path : '.'
```
Comment (mheap) [inline lib/nexmo_markdown_renderer.rb]: I'd expect this to use the following options:

* `Rails.configuration.docs_base_path`
* `ENV['DOCS_BASE_PATH']`
* `.`

Comment (hummusonrails) [reply lib/nexmo_markdown_renderer.rb]: 👍 100% I'll fix that.

File: lib/nexmo_markdown_renderer.rb [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # frozen_string_literal: true
 GEM_ROOT = File.expand_path("../..", __FILE__)
+if (defined?(Rails) && Rails.configuration.docs_base_path)
```
Comment (fabianrbz) [inline lib/nexmo_markdown_renderer.rb]:
```suggestion
  DOCS_BASE_PATH  = defined?(Rails) && Rails.configuration.docs_base_path || ENV.fetch('DOCS_BASE_PATH', '.')
```

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -20,8 +20,8 @@
 
 RSpec.configure do |config|
   config.before(:each) do
-    ENV['DOCS_BASE_PATH'] = 'spec/fixtures'
-    @original_dictionary =Nexmo::Markdown::DocFinder .dictionary
+    # DOCS_BASE_PATH = 'spec/fixtures'
```
Comment (fabianrbz) [inline spec/spec_helper.rb]: can you remove this comment?

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -20,8 +20,8 @@
...
+    @original_dictionary =Nexmo::Markdown::DocFinder.dictionary
```
Comment (fabianrbz) [inline spec/spec_helper.rb]: nitpick, add a space after the `=`

**Review by fabianrbz — CHANGES_REQUESTED** [review]
i left a few comments

**Comment by hummusonrails** [comment]
> i left a few comments

Addressed your review in the latest commit. Thanks, @fabianrbz!

File: lib/nexmo_markdown_renderer/filters/code_filter.rb [hunk 5]
```diff
@@ -6,12 +6,12 @@ def call(input)
           config = YAML.safe_load($1)
     
           if config['config']
-            configs = YAML.load_file("#{ENV['DOCS_BASE_PATH']}/config/code_examples.yml")
+            configs = YAML.load_file("#{Nexmo::Markdown::Config.Nexmo::Markdown::Config.docs_base_path}/config/code_examples.yml")
```
Comment (fabianrbz) [inline lib/nexmo_markdown_renderer/filters/code_filter.rb]: @benhayehudi i think you have one extra `Nexmo::Markdown::Config`

File: lib/nexmo_markdown_renderer.rb [hunk 6]
```diff
@@ -6,6 +6,7 @@
 require 'open-uri'
 require 'active_model'
 require 'i18n'
+require_relative 'nexmo_markdown_renderer/config/load_config'
```
Comment (fabianrbz) [inline lib/nexmo_markdown_renderer.rb:9]: @benhayehudi looks like you forgot to commit the file.

**Review by fabianrbz — CHANGES_REQUESTED** [review]
left a few comments, but overall it looks good

File: lib/nexmo_markdown_renderer/filters/code_filter.rb (hunk 5 above)
Comment (hummusonrails) [reply lib/nexmo_markdown_renderer/filters/code_filter.rb]: fixed!

Comment (fabianrbz) [inline lib/nexmo_markdown_renderer/filters/code_filter.rb]: @benhayehudi this still looks odd

Comment (hummusonrails) [reply lib/nexmo_markdown_renderer/filters/code_filter.rb]: Please check again now. 🙏

**Review by fabianrbz — APPROVED** [review]

**Pull request merged**